torch/distributed/fsdp/_flat_param.py

```diff
@@ -27,6 +27,7 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from torch import Tensor
+from torch.distributed._tensor import DTensor
```

Comment:
This is not a great state to be in, but I always remember that we cannot import `DTensor` at the top-level of this file, or else we may break some internal torch package or torch deploy thing.

I am not too familiar with the issue though :/

Reply:
Is the breakage caused by circular dependency, if you can recall?

Reply:
I honestly cannot remember :(

Reply:
got you. will import DTensor inside the function